### PR TITLE
✋ Fix single asset page showing other tokens

### PR DESCRIPTION
**Pull request by henryboldi**
Closes https://github.com/tallycash/extension/issues/588

Heavily heavily based on Antonio's code here https://github.com/tallycash/extension/issues/588#issuecomment-991819719

### CODE REVIEW

File: ui/pages/SingleAsset.tsx [hunk 1]
```diff
@@ -24,13 +24,29 @@ export default function SingleAsset(): ReactElement {
...
+        if (activity.annotation?.type === "asset-transfer") {
+          return activity.annotation.assetAmount.asset.symbol === symbol
+        }
+        if (activity.annotation?.type === "asset-swap") {
+          return (
+            activity.annotation.fromAssetAmount.asset.symbol === symbol ||
+            activity.annotation.toAssetAmount.asset.symbol === symbol
+          )
+        }
+        if (activity.annotation?.type === "contract-interaction") {
```
Comment (Shadowfiend) [inline ui/pages/SingleAsset.tsx:45]: We should handle approvals as well. Probably should `switch` on the type so we get an exhaustiveness warning tbh, but one thing at a time 😅 I do think include approvals here to minimize confusion though.

**Review by Shadowfiend — APPROVED** [review]
I'm good here, but need @mhluongo (or anyone) to confirm with a test drive as not at a machine.

**Comment by mhluongo** [comment]
Screenshot from 2021-12-15 21-53-52

:heart_eyes:

**Comment by mhluongo** [comment]
This does show random contract interactions as ETH activities though :thinking:

**Comment by mhluongo** [comment]
Screenshot from 2021-12-15 21-57-03

**Comment by mhluongo** [comment]
Merging and [opening an issue for discussion](https://github.com/tallycash/extension/issues/676)

**Pull request merged**